### PR TITLE
LPS-162827 Add Actions to the Version History Tests

### PR DESCRIPTION
**Pull request by jkappler**
## Motivation

[Link to task](https://issues.liferay.com/browse/LPS-164363)

This PR contains unit and integration tests for actions that are shown for versions in info panel in Content Dashboard

## Change summary:

* Unit tests for ContentDashboardItemVersionActions (DM and Web Content)
* Integration tests for ExpireJournalArticleContentDashboardItemVersionActionProvider
* Integration tests for ViewFileVersionContentDashboardItemVersionActionProvider
* Integration test for ContentDashboardItemVersionActionProviderTracker

### CODE REVIEW

**Comment by jkappler** [comment]
ci:test:relevant

**Comment by jkappler** [comment]
ci:test:sf

File: modules/apps/content-dashboard/content-dashboard-document-library-impl/src/test/java/com/liferay/content/dashboard/document/library/internal/item/FileEntryContentDashboardItemTest.java [hunk 1]
```diff
@@ -375,6 +518,26 @@ public void testIsViewable() throws Exception {
 				_getHttpServletRequest(RandomTestUtil.randomLong())));
 	}
 
+	private void _assertContentDashboardItemVersionAction(
+		ContentDashboardItemVersionAction
```
Comment (cgoncas) [inline modules/apps/content-dashboard/content-dashboard-document-library-impl/src/test/java/com/liferay/content/dashboard/document/library/internal/item/FileEntryContentDashboardItemTest.java:522]: Hey @jkappler !!

Can we use the equals method in ContentDashboardItemVersionAction for this?

Thanks!!

File: modules/apps/content-dashboard/content-dashboard-document-library-test/src/testIntegration/java/com/liferay/content/dashboard/document/library/internal/item/action/provider/test/ViewFileVersionContentDashboardItemVersionActionProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+			"view", contentDashboardItemVersionAction.getName());
+		Assert.assertEquals(
+			"view", contentDashboardItemVersionAction.getIcon());
+		Assert.assertEquals(
+			_language.get(LocaleUtil.US, "view"),
+			contentDashboardItemVersionAction.getLabel(LocaleUtil.US));
+
+		String url = contentDashboardItemVersionAction.getURL();
+
+		Assert.assertTrue(Validator.isNotNull(url));
```
Comment (cgoncas) [inline modules/apps/content-dashboard/content-dashboard-document-library-test/src/testIntegration/java/com/liferay/content/dashboard/document/library/internal/item/action/provider/test/ViewFileVersionContentDashboardItemVersionActionProviderTest.java:117]: Maybe we can use Assert.assetNotNull?

File: modules/apps/content-dashboard/content-dashboard-document-library-test/src/testIntegration/java/com/liferay/content/dashboard/document/library/internal/item/action/provider/test/ViewFileVersionContentDashboardItemVersionActionProviderTest.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+			url.contains("fileEntryId=" + _fileEntry.getFileEntryId()));
+		Assert.assertTrue(
+			url.contains(
+				"mvcRenderCommandName=" +
+					HtmlUtil.escapeURL("/document_library/view_file_entry")));
+		Assert.assertTrue(url.contains("version=" + _fileEntry.getVersion()));
+	}
+
+	@Test
+	public void testIsShow() throws Exception {
```
Comment (cgoncas) [inline modules/apps/content-dashboard/content-dashboard-document-library-test/src/testIntegration/java/com/liferay/content/dashboard/document/library/internal/item/action/provider/test/ViewFileVersionContentDashboardItemVersionActionProviderTest.java:129]: What do you think about adding a method to test when this action is not show?

File: modules/apps/content-dashboard/content-dashboard-journal-impl/src/test/java/com/liferay/content/dashboard/journal/internal/item/JournalArticleContentDashboardItemTest.java [hunk 4]
```diff
@@ -19,17 +19,25 @@
 import com.liferay.content.dashboard.item.ContentDashboardItemVersion;
```
Comment (cgoncas) [inline modules/apps/content-dashboard/content-dashboard-journal-impl/src/test/java/com/liferay/content/dashboard/journal/internal/item/JournalArticleContentDashboardItemTest.java:19]: How do you feel about moving all the unit tests to integration tests?

I think as we don't need to use mocks maybe we could read them more easily?

File: modules/apps/content-dashboard/content-dashboard-journal-test/src/testIntegration/java/com/liferay/content/dashboard/journal/internal/item/action/provider/test/ExpireJournalArticleContentDashboardItemVersionActionProviderTest.java [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+				mockHttpServletRequest, LocaleUtil.US,
+				TestPropsValues.getUser()));
+
+		Assert.assertTrue(
+			_contentDashboardItemVersionActionProvider.isShow(
+				journalArticle, mockHttpServletRequest));
+	}
+
+	@Test
+	public void testIsShowExpiredArticle() throws Exception {
```
Comment (cgoncas) [inline modules/apps/content-dashboard/content-dashboard-journal-test/src/testIntegration/java/com/liferay/content/dashboard/journal/internal/item/action/provider/test/ExpireJournalArticleContentDashboardItemVersionActionProviderTest.java:140]: I love this tests, awesome work :)

**Review by cgoncas — APPROVED** [review]
I have added some comments, any of them needs to be a stopper to send it to Brian.

File: modules/apps/content-dashboard/content-dashboard-document-library-impl/src/test/java/com/liferay/content/dashboard/document/library/internal/item/FileEntryContentDashboardItemTest.java (hunk 1 above)
Comment (jkappler) [reply modules/apps/content-dashboard/content-dashboard-document-library-impl/src/test/java/com/liferay/content/dashboard/document/library/internal/item/FileEntryContentDashboardItemTest.java:522]: Will change this @cgoncas!

File: modules/apps/content-dashboard/content-dashboard-document-library-test/src/testIntegration/java/com/liferay/content/dashboard/document/library/internal/item/action/provider/test/ViewFileVersionContentDashboardItemVersionActionProviderTest.java (hunk 2 above)
Comment (jkappler) [reply modules/apps/content-dashboard/content-dashboard-document-library-test/src/testIntegration/java/com/liferay/content/dashboard/document/library/internal/item/action/provider/test/ViewFileVersionContentDashboardItemVersionActionProviderTest.java:117]: My bad @cgoncas, will also change this

File: modules/apps/content-dashboard/content-dashboard-document-library-test/src/testIntegration/java/com/liferay/content/dashboard/document/library/internal/item/action/provider/test/ViewFileVersionContentDashboardItemVersionActionProviderTest.java (hunk 3 above)
Comment (jkappler) [reply modules/apps/content-dashboard/content-dashboard-document-library-test/src/testIntegration/java/com/liferay/content/dashboard/document/library/internal/item/action/provider/test/ViewFileVersionContentDashboardItemVersionActionProviderTest.java:129]: @cgoncas, it's always shown, that's why I only created one test

Comment (cgoncas) [reply modules/apps/content-dashboard/content-dashboard-document-library-test/src/testIntegration/java/com/liferay/content/dashboard/document/library/internal/item/action/provider/test/ViewFileVersionContentDashboardItemVersionActionProviderTest.java:129]: awesoime

File: modules/apps/content-dashboard/content-dashboard-journal-impl/src/test/java/com/liferay/content/dashboard/journal/internal/item/JournalArticleContentDashboardItemTest.java (hunk 4 above)
Comment (cgoncas) [reply modules/apps/content-dashboard/content-dashboard-journal-impl/src/test/java/com/liferay/content/dashboard/journal/internal/item/JournalArticleContentDashboardItemTest.java:19]: Me too :).

**Comment by jkappler** [comment]
ci:stop

File: modules/apps/content-dashboard/content-dashboard-document-library-test/src/testIntegration/java/com/liferay/content/dashboard/document/library/internal/item/action/provider/test/ViewFileVersionContentDashboardItemVersionActionProviderTest.java (hunk 2 above)
Comment (jkappler) [reply modules/apps/content-dashboard/content-dashboard-document-library-test/src/testIntegration/java/com/liferay/content/dashboard/document/library/internal/item/action/provider/test/ViewFileVersionContentDashboardItemVersionActionProviderTest.java]: @cgoncas, added a new commit to standardize all tests https://github.com/liferay-tango/liferay-portal/pull/2917/commits/7ab79ec3d7e954efbb508b4da094c190ef7d7f71. Most likely we were checking for empty strings, but the next lines should always catch this

**Comment by jkappler** [comment]
ci:test:sf

**Comment by jkappler** [comment]
ci:test:relevant

**Comment by jkappler** [comment]
@cgoncas, will create the task to move unit tests to integration tests ❤️

**Comment by jkappler** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:forward

**Pull request closed without merging**